### DISCUSSION TITLE
auth.update() password returns "Error: Not logged in."

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [supabase/supabase#5887] [Questions]
I have a Vue3 app with a decoupled backend using Apollo Server. On my frontend, I log in to the app via Supabase, then navigate to the Profile Settings page where a user can update their password. In my `updatePassword` resolver, I run:

```typescript
import { UpdatePasswordInput, UpdatePasswordPayload } from '@/generated/types';
import supabase from '@/lib/supabase';
import { genericError } from '@/helpers/errors';

const updatePassword = async (_parent: any, { input }: { input: UpdatePasswordInput }): Promise<UpdatePasswordPayload> => {
  const { password } = input;

  if (!password) {
    return {
      success: false, ...genericError({ message: 'New password not provided', path: 'updatePassword resolver' })
    };
  }

  try {
    const { user, error } = await supabase.auth.update({ password });
    console.log({ user });
    console.log({ error }); // <-- returns "Error: Not logged in".

    if (error || !user) {
      return {
        success: false, ...genericError({ message: 'Error updating password', path: 'updatePassword resolver: try' })
      };
    }

    return {
      success: true,
      errors: null,
    };
  } catch (err) {
    console.log({ err });
    return {
      success: false, ...genericError({ message: 'Error updating password', path: 'updatePassword resolver: catch' })
    };
  }
};

export default updatePassword;
```

The line where I'm calling `supabase.auth.update({ password });` is returning the error, telling me that the user is not logged in. I'm sure it's because of the fact that the session is being stored on the frontend, and digging in to the .update() method in GoTrueClient, I can see that the update method is in fact checking for `this.currentSession?.access_token`. 

How can I go about updating the user password, with a pair of apps that are decoupled? Or is this a much larger issue in the way I'm handling login?

**Comment by maxkozlowski** [comment]
Have you found the workaround in the end? I'm having a similar issue.

**Comment by nuno84** [comment]
Same here, on Flutter. Any tip?
Thanks

**Reply by nuno84** [reply]
I got it now.
For anyone that gets the same issue, here is the reason I found.

In my case, it was because for some reason I don't understand, Flutter by default adds a # sign automatically after the domain:
`https://myawesomeapp.com/#`

And after the redirect it uses # again for the parameters (token, etc). 

`Something like: https://myawesomeapp.com/#/recover#token=blablabla`

Then supabase auth parser gets it wrong because it parses from the first # character and doesnt get the token, so there is no session available.
I added the 'url_strategy' package to stop that # sign from showing up in the base app URL and it works fine now.

**Comment by versoworks** [comment]
> How can I go about updating the user password, with a pair of apps that are decoupled? Or is this a much larger issue in the way I'm handling login?

I'm not familiar with Vue3, but it sounds like you're wanting to update a user on the server side. If so, you'll need to create a Supabase client with your `service role` key and call [`updateuserbyid`](https://supabase.com/docs/reference/javascript/auth-admin-updateuserbyid).
